test/02_test/test/spam_test.py

```diff
@@ -43,3 +45,10 @@ def test_virtualenv(self):
             print("=[listdir]2", os.listdir(os.path.join(virtualenv_path, 'bin')))
         self.assertTrue(path_contains(virtualenv_path, sys.executable))
         self.assertTrue(path_contains(virtualenv_path, spam.__file__))
+
+    def test_uname(self):
+        if platform.system() == "Windows":
+            return
+        bits = struct.calcsize("P") * 8
+        if bits == 32:
+            self.assertEqual(os.uname()[4], "i686")
```

Comment:
platform.machine() will do the trick - it uses the same data as os.uname()

Reply:
It is somehow documented?

Reply:
https://github.com/python/cpython/blob/b88cd585d36d6285a5aeb0b6fdb70c134062181e/Lib/platform.py#L935

It literally calls `uname().machine`

Reply:
I guess just writing `uname().machine` is also fine, then, here. But I'd replace the `[4]`, at least.

Reply:
> https://github.com/python/cpython/blob/b88cd585d36d6285a5aeb0b6fdb70c134062181e/Lib/platform.py#L935
> 
> It literally calls `uname().machine`

This is implementation, not documentation. So it can change in future. 

> I guess just writing `uname().machine` is also fine, then, here. But I'd replace the `[4]`, at least.

Not work on python 2.7 https://docs.python.org/2.7//library/os.html#os.uname

Reply:
This is as good as I can do - https://github.com/python/cpython/blob/2.7/Lib/platform.py#L1328

Reply:
Ok. Changed.

Reply:
> Not work on python 2.7 https://docs.python.org/2.7//library/os.html#os.uname

_Grrrrrrm, **sigh**_